.github/workflows/all.yml

```diff
@@ -0,0 +1,319 @@
+name: Deps, Lint, Test, Build, Publish
+on:
+  # Enable manually triggering this workflow via the API or web UI
+  workflow_dispatch:
+  push:
+    branches:
+      - master
+    tags:
+      - v*
+  pull_request:
+
+defaults:
+  run:
+    shell: bash
+
+env:
+  APP_NAME: "k6"
+  DOCKER_IMAGE_ID: ${{ github.repository }}
+
+jobs:
+  deps:
+    runs-on: ubuntu-latest
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.14.x
+      - name: Check dependencies
+        run: |
+            go version
+            test -z "$(go mod vendor && git status --porcelain)"
+            go mod verify
+
+  lint:
+    runs-on: ubuntu-latest
+    env:
+        GOLANGCI_VERSION: v1.31
+        GO111MODULE: 'on'
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+        with:
+          fetch-depth: 0
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: 1.14.x
+      - name: Install golangci-lint
+        working-directory: /tmp
+        run: go get github.com/golangci/golangci-lint/cmd/golangci-lint@$GOLANGCI_VERSION
+      - name: Run linters
+        run: |
+          BASEREV=$(git merge-base HEAD origin/master)
+          echo "Base revision: $BASEREV"
+          golangci-lint run --out-format=tab --new-from-rev "$BASEREV" ./...
+
+  test:
+    strategy:
+      matrix:
+        go-version: [1.14.x]
+        platform: [ubuntu-latest, windows-latest]
+    runs-on: ${{ matrix.platform }}
+    steps:
+      - name: Checkout code
+        uses: actions/checkout@v2
+      - name: Install Go
+        uses: actions/setup-go@v2
+        with:
+          go-version: ${{ matrix.go-version }}
+      - name: Run tests
+        run: |
+          set -x
+          go version
+          export GOMAXPROCS=2
+          args=("-race")
```

Comment:
Did you investigate whether `-p 2` (or any other number) is faster? `go test -p` actually does different things than `GOMAXPROCS`.

Reply:
hmm maybe `-p 2` could also fix the MacOS flakiness? 
```
	-p n
		the number of programs, such as build commands or
		test binaries, that can be run in parallel.
		The default is the number of CPUs available.
```

Reply:
Yeah, I was confusing `go test -parallel` (which defaults to `GOMAXPROCS`), with `go test -p`, so I thought it was redundant... Damn these non-POSIX flags!

OK, I'll try running macOS and Windows with `GOMAXPROCS=1`, `-p 1` and no `-race`.

Reply:
No, even with `-p 1` tests on macOS are very time sensitive and [fail consistently](https://github.com/imiric/k6/runs/1225385747). I'll still leave it for Windows as it should make them more reliable.

Reply:
it seems this is a [known issue](https://github.com/actions/virtual-environments/issues/1336) that isn't getting better :(

Reply:
:disappointed: the sad fact is that a lot of our tests are too time sensitive, but we can't rewrite them quickly... I still think _some_ MacOS tests are better than none, so maybe we should check if we're running on a mac and `Skip()` some tests, like we do for Windows: https://github.com/loadimpact/k6/search?q=goos+windows

Then we can create an issue like this one for Mac: https://github.com/loadimpact/k6/issues/902

btw I know some of the reasons we skip some tests on Windows were caused specifically by AppVeyor, not Windows itself. So we might unskip some of the currently skipped tests after we merge this! :crossed_fingers:

Reply:
I propose that we do any kind of CI improvements after we merge this PR, not in this PR ;)

Reply:
hmm fair enough, enabling partial macos tests and more windows tests is better done afterwards :+1: